File: python/structural/flyweight/map/icon.py
class Icon:
    def __init__(self, type: str, image_path: str):
        self.type = type               
        self.image_path = image_path    

    def render(self, latitude: float, longitude: float):
        print(f"📍 Render '{self.type.upper()}' at ({latitude}, {longitude}) using image: {self.image_path}")

File: typescript/src/structural/flyweight/map/Icon.ts
export class Icon {
    private type: string;
    private imagePath: string;

    constructor(type: string, imagePath: string){
        this.type = type;
        this.imagePath = imagePath;
    }

    static render(icon: Icon, latitude: number, longitude: number): void{
        console.log(`Render icon type '${icon.type.toUpperCase()}' at (${latitude} - ${longitude}) using image: ${icon.imagePath}`)
    }
}